### PR TITLE
Closes #714: License Difference on Submitting Dataset for Review

### LINKED ISSUE #714: When sending to review user should see table with changed licenses from latest version

### PR DESCRIPTION
**Pull request by dbojanek**

### CODE REVIEW

File: dataverse-webapp/src/test/java/edu/harvard/iq/dataverse/dataset/difference/LicenseDifferenceFinderTest.java [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+        assertSame(v2File1Terms, termsDiffs.get(0).getNewTerms());
+
+        assertEquals(String.valueOf(dataFile2.getId()), termsDiffs.get(1).getFileSummary().getFileId());
+        assertSame(v1File2Terms, termsDiffs.get(1).getOldTerms());
+        assertSame(v2File2Terms, termsDiffs.get(1).getNewTerms());
+    }
+
+    private DatasetVersion parseDatasetVersionFromClasspath(String classpath) throws IOException, JsonParseException {
+
+        try (ByteArrayInputStream is = new ByteArrayInputStream(IOUtils.resourceToByteArray(classpath))) {
```
Comment (MrK191) [inline dataverse-webapp/src/test/java/edu/harvard/iq/dataverse/dataset/difference/LicenseDifferenceFinderTest.java]: We can use UnitTestUtils for that now.

File: dataverse-webapp/src/test/java/edu/harvard/iq/dataverse/dataset/difference/LicenseDifferenceFinderTest.java [hunk 2]
```diff
@@ -0,0 +1,264 @@
...
+
+        when(datasetFieldService.findByNameOpt(eq("depositor"))).thenReturn(depositorFieldType);
+        when(datasetFieldService.findByNameOpt(eq("dateOfDeposit"))).thenReturn(dateOfDepositFieldType);
+
+
+        dataset = MocksFactory.makeDataset();
+    }
+
+    @Test
+    public void create__WITH_CHANGED_FILE_TERMS() throws IOException, JsonParseException {
```
Comment (MrK191) [inline dataverse-webapp/src/test/java/edu/harvard/iq/dataverse/dataset/difference/LicenseDifferenceFinderTest.java]: maybe just "getLicenseDifference_...".

File: dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/dataset/difference/LicenseDifferenceFinder.java [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+    }
+
+    private List<Tuple2<String, TermsOfUseDiff>> getTermsOfUseDiffs(List<FileMetadata> newVersionFileMetadata, List<FileMetadata> previousVersionFileMetadata) {
+        List<Tuple2<String, TermsOfUseDiff>> changedTermsOfUse = Lists.newArrayList();
+
+        for (FileMetadata fmdo : previousVersionFileMetadata) {
+            for (FileMetadata fmdn : newVersionFileMetadata) {
+                if (fmdo.getDataFile().equals(fmdn.getDataFile())) {
+                    if (!areFileTermsEqual(fmdo.getTermsOfUse(), fmdn.getTermsOfUse())) {
+                        changedTermsOfUse.add(new Tuple2<>(fmdn.getLabel(), new TermsOfUseDiff(fmdo.getTermsOfUse(), fmdn.getTermsOfUse())));
```
Comment (MrK191) [inline dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/dataset/difference/LicenseDifferenceFinder.java]: Hmm, shouldn't the method to create "DatasetFileTermDifferenceItem" just be called here, and this method could return List<DatasetFileTermDifferenceItem>. I don't see much benefit in doing List\<Tuple> ->  List\<DatasetFileTermDifferenceItem>.

**Review by MrK191 — CHANGES_REQUESTED** [review]

File: dataverse-webapp/src/test/java/edu/harvard/iq/dataverse/dataset/difference/LicenseDifferenceFinderTest.java (hunk 2 above)
Comment (dbojanek) [reply dataverse-webapp/src/test/java/edu/harvard/iq/dataverse/dataset/difference/LicenseDifferenceFinderTest.java]: Tbh all tests have been taken directly from `datasetversiondifferencetest` and I've only limited logic to `LicenseDifference` :)

File: dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/dataset/difference/LicenseDifferenceFinder.java (hunk 3 above)
Comment (dbojanek) [reply dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/dataset/difference/LicenseDifferenceFinder.java]: This approach was an artifact from earlier. Luckily you've spotted that :+1:

File: dataverse-webapp/src/test/java/edu/harvard/iq/dataverse/dataset/difference/LicenseDifferenceFinderTest.java (hunk 1 above)
Comment (dbojanek) [reply dataverse-webapp/src/test/java/edu/harvard/iq/dataverse/dataset/difference/LicenseDifferenceFinderTest.java]: What's that?

Comment (MrK191) [reply dataverse-webapp/src/test/java/edu/harvard/iq/dataverse/dataset/difference/LicenseDifferenceFinderTest.java]: It's just a wrapper for getting stuff from resources, we can also control which library we are going to use globally then.

**Review by MrK191 — APPROVED** [review]

**Pull request merged**